### PR TITLE
Make testing equivalence of floats in SITL a compilation error

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
The last patch here has me a little dubious; should we really be using an epsilon?

Perhaps we should instead pragma-off the compiler error for test_math.cpp?

File: libraries/SITL/SIM_XPlane.cpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ bool XPlane::receive_data(void)
                  */
                 bool has_magic = ((uint32_t)(data[1] * throttle_magic_scale) % 1000U) == (uint32_t)(throttle_magic * throttle_magic_scale);
                 if (data[1] < 0 ||
-                    data[1] == throttle_sent ||
+                    is_equal(data[1], throttle_sent) ||
```
Comment (tridge) [inline libraries/SITL/SIM_XPlane.cpp]: no, this really, really needs to be ==, a epsilon will break this

**Comment by peterbarker** [comment]
Assigned devcalltopic in case somebody on the call knows the tests should be strict-equivalence.

**Comment by peterbarker** [comment]
It was decided on the devcall that the math tests should be allowed to directly compare floats for equivalence, on the basis that they should know what they're doing.

Removed old changes, pragma'd away the `-Wfloat-equal`.

**Pull request merged**